File: app/src/modules/calendar/api/data-transfer-objects/update-appointment.ts
import { CreateAppointmentBlock } from '@/modules/calendar/api/data-transfer-objects/create-appointment-block';

export type UpdateAppointment = {
    id: string;
    serviceId: string;
    price: number;
    blocks: CreateAppointmentBlock[];
    clientId: string;
    notes?: string;
};

File: api/src/Domain/Calendar/DataTransferObjects/UpdateAppointment.cs
using System;
using System.Collections.Generic;

namespace DetailingArsenal.Domain.Calendar {
    public class UpdateAppointment : IDataTransferObject {
        public Guid ServiceId { get; }
        public Guid ClientId { get; }
        public decimal Price { get; }
        public string? Notes { get; }
        public List<UpdateAppointmentBlock> Blocks { get; set; } = new List<UpdateAppointmentBlock>();

        public UpdateAppointment(Guid serviceId, Guid clientId, decimal price, string? notes) {
            ServiceId = serviceId;
            ClientId = clientId;
            Price = price;
            Notes = notes;
        }
    }
}